scipy/stats/tests/test_morestats.py

```diff
@@ -2595,9 +2595,17 @@ def test_directional_stats_correctness(self):
         reference_mean = np.array([0.2984, -0.1346, -0.9449])
         assert_allclose(mean_rounded, reference_mean)
 
-        expected_var = 0.025335389565304012
-        directional_var = 1 - dirstats.mean_resultant_length
-        assert_allclose(expected_var, directional_var)
+    @pytest.mark.parametrize('angles, ref', [
+        ([-np.pi/2, np.pi/2], 1.),
+        ([0, 2*np.pi], 0.),
+        ([-np.pi/2, -np.pi/4, 0, np.pi/4, np.pi/2], stats.circvar)
+    ])
```

Comment:
To make the tests more robust, I could add one more testcase that generates angles randomly and tests against `circvar`. Let me know if such a test would be helpful and I will add it.

Reply:
Isn't that what `test_directional_stats_2d` does?

Reply:
Oh, yes! I didn't see the test. Do you want me to remove the last testcase. Since the first two test for special cases, I think we should keep them. If you want, I can merge those with `test_directional_stats_2d`.

Reply:
You can definitely leave out the last test case. If they can all be combined cleanly, sure. 
Please merge (not rebase, of course) main while you're at it. Maybe consider gh-17308 before pushing so you get greener CI.

Reply:
I removed the last testcase and merged main. If you are happy with it, we can merge once the CI's green.